src/Microsoft.TemplateEngine.Edge/EngineEnvironmentSettings.cs

```diff
@@ -14,52 +14,42 @@
 
 namespace Microsoft.TemplateEngine.Edge
 {
-    public sealed class EngineEnvironmentSettings : IEngineEnvironmentSettings, IDisposable
+    public sealed class EngineEnvironmentSettings : IEngineEnvironmentSettings
     {
-        private volatile bool _disposed;
-
         public EngineEnvironmentSettings(
             ITemplateEngineHost host,
             bool virtualizeSettings = false,
-            Action<IEngineEnvironmentSettings>? onFirstRun = null,
             string? settingsLocation = null,
-            IEnvironment? environment = null)
+            IEnvironment? environment = null,
+            IComponentManager? componentManager = null,
+            IPathInfo? pathInfo = null)
         {
             Host = host ?? throw new ArgumentNullException(nameof(host));
             Environment = environment ?? new DefaultEnvironment();
-            SettingsLoader = new SettingsLoader(this, onFirstRun);
-            Paths = new DefaultPathInfo(this, settingsLocation);
+            Paths = pathInfo ?? new DefaultPathInfo(this, settingsLocation);
             if (virtualizeSettings)
             {
                 Host.VirtualizeDirectory(Paths.GlobalSettingsDir);
             }
+            Components = componentManager ?? new ComponentManager(this);
         }
 
-        public EngineEnvironmentSettings(ITemplateEngineHost host, ISettingsLoader settingsLoader, IPathInfo pathInfo, IEnvironment? environment)
+        [Obsolete("ISettingsLoader is obsolete, see ISettingsLoader obsolete description for details.")]
+        public ISettingsLoader SettingsLoader
         {
-            Host = host ?? throw new ArgumentNullException(nameof(host));
-            SettingsLoader = settingsLoader ?? throw new ArgumentNullException(nameof(settingsLoader));
-            Paths = pathInfo ?? throw new ArgumentNullException(nameof(pathInfo));
-            Environment = environment ?? new DefaultEnvironment();
+            get
+            {
+                throw new Exception("ISettingsLoader is no longer supported.");
```

Comment:
We marked this as obsolete, so that people referencing this will have some time to update their code before we remove this and break them. But then we throw here and we break them starting the first day anyway. I'm wondering if it would be better to:
- Remove this property all together without marking it obsolete first. 
Or
- Return null in the getter so that hopefully some users won't be broken (?).

Reply:
We want to keep it so API users can see Obsolete message...
We want to throw exception so they see this message that they can't use this anymore...
We don't want to invest work into proxying all SettingsLoaders calls just to make it compatible for few users of this API for .NET 6, too much already changed...